internal/wal/wal_aof.go

```diff
@@ -18,43 +20,146 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-var writeBuf bytes.Buffer
+const (
+	segmentPrefix = "segment-"
+)
+
+var (
+	bufferPool = sync.Pool{
+		New: func() interface{} {
+			return make([]byte, 0, 4096) // Pre-allocate 4KB buffer
+		},
+	}
+)
 
 type WALAOF struct {
-	file   *os.File
-	mutex  sync.Mutex
-	logDir string
+	mutex               sync.Mutex
+	syncMutex           sync.Mutex    // Separate mutex for syncing
+	file                *os.File      // Current WAL file
+	writer              io.Writer     // Writer interface, can be buffered or unbuffered
+	bufWriter           *bufio.Writer // Buffered writer if buffering is enabled
+	logDir              string
+	syncInterval        time.Duration // Interval for syncing data to disk
+	lastSyncTime        time.Time     // Last time data was synced
+	shouldFsync         bool
```

Comment:
this is the write mode and not "shouldFsync".

Reply:
This was just a POC addition. I will revamp the WALEntry and configs right now